modules/serverless-apiserversource.adoc

```diff
@@ -0,0 +1,231 @@
+====
++
+Create the service account by entering the following command:
++
+----
+$ oc apply --filename serviceaccount.yaml
+----
+
+. Create an ApiServerSource event source.
++
+You can do this by creating a file named `k8s-events.yaml` and copying the following sample code into it:
++
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+  namespace: default
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1beta1
+      kind: Broker
+      name: default
+----
++
+[NOTE]
+====
+If you want to consume events from a different namespace or use a different Service Account, you need to modify the sample `k8s-events.yaml` file accordingly.
+====
++
+Create the event source by entering the following command:
++
+----
+$ oc apply --filename k8s-events.yaml
+----
+
+. To check that the ApiServerSource is set up correctly, create a Knative service that dumps incoming messages to its log.
++
+Copy the following sample YAML into a file named `service.yaml`:
++
+----
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+  namespace: default
+spec:
+  template:
+    spec:
+      containers:
+        - image: quay.io/openshift-knative/knative-eventing-sources-event-display
+----
++
+Create the service by entering the following command:
++
+----
+$ oc apply --filename service.yaml
+----
+
+. Create a trigger from the `default` broker to filter events to the service created in the previous step.
++
+You can create the trigger by entering the following command:
++
+----
+$ kn trigger create foo --broker default --sink svc:event-display \
+--filter source=dev.knative.samples/testevents \
+----
++
+For more information about creating and using triggers with `kn`, see the documentation on xref:../../serverless/knative_eventing/serverless-kn-trigger.adoc#serverless-kn-trigger[Using triggers with Knative Eventing].
++
+Alternatively, you can create a file named `trigger.yaml` and copy the sample code below into it:
++
+----
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: testevents-trigger
+  namespace: default
+spec:
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+----
++
+After you have created the `trigger.yaml` file, apply it by entering the following command:
++
+----
+$ oc apply --filename trigger.yaml
+----
+. Create events by launching a pod in the default namespace.
+
+.Verification steps
+
+You can verify that the Kubernetes events were sent into the Knative eventing system by looking at the message dumper function logs.
+
+You can view the message dumper function logs by entering the following commands:
+----
+$ oc get pods
+$ oc logs -l serving.knative.dev/service=event-display -c user-container
```

Comment:
Are you sure that `oc logs` supports the `-l` option ? I tried it here with oc v4.3.8 and it doesn't know how to select a pod by label (in contrast to kubectl which does it):

```
 oc logs  -l serving.knative.dev/service=event-display -c user-container
error: You must provide one or more resources by argument or filename.
Example resource specifications include:
   '-f rsrc.yaml'
   '--filename=rsrc.json'
   '<resource> <name>'
   '<resource>'
```

I works with kubectl, though.

Strange enough, `oc logs --help` shows that there is a `-l` option, but it apparently doesn't work.

Reply:
Interesting - I have no idea if it works. This came from upstream docs that use `kubectl`, so the assumption we make in most cases is that we can change that to `oc` for our downstream docs and it should work.

Is there a command we can use for `oc` that does the same thing if this doesn't work?

Reply:
That's a good question. But let's first confirm that `oc logs -l`  really doesn't work. I will take that over to slack to discuss this.

Reply:
I opened https://github.com/openshift/oc/issues/365 to track this issue, but in the meantime you could also use

```
oc logs $(oc get pod -o name | grep event-display) -c user-container
```